src/streamlink_cli/output.py

```diff
@@ -108,13 +109,27 @@ def _create_arguments(self):
             filename = self.http.url
         else:
             filename = "-"
-
         args = self.args.format(filename=filename)
         cmd = self.cmd
+        extra_args = []
+
+        if self.title is not None:
+            # vlc
+            if self.cmd in SUPPORTED_PLAYERS['vlc']:
```

Comment:
There is a problem with this, it doesn't work if you have some extra args with `-p`, for example `-p "mpv --no-audio"`. However, I have a fix ;)

```diff
diff --git a/src/streamlink_cli/output.py b/src/streamlink_cli/output.py
index 31af016..14fc797 100644
--- a/src/streamlink_cli/output.py
+++ b/src/streamlink_cli/output.py
@@ -82,6 +82,7 @@ class PlayerOutput(Output):
         self.namedpipe = namedpipe
         self.http = http
         self.title = title
+        self.player = self.supported_player(self.cmd)

         if self.namedpipe or self.filename or self.http:
             self.stdin = sys.stdin
@@ -100,6 +101,13 @@ class PlayerOutput(Output):
         sleep(0.5)
         return self.player.poll() is None

+    @staticmethod
+    def supported_player(cmd):
+        """
+        Check if the current player supports adding a title
+
+        :param cmd: cmd to test
+        :return: name of the player|None
+        """
+        for player, possiblecmds in SUPPORTED_PLAYERS.items():
+            for possiblecmd in possiblecmds:
+                if possiblecmd in cmd:
+                    return player
+
     def _create_arguments(self):
         if self.namedpipe:
             filename = self.namedpipe.path
@@ -115,13 +123,13 @@ class PlayerOutput(Output):

         if self.title is not None:
             # vlc
-            if self.cmd in SUPPORTED_PLAYERS['vlc']:
+            if self.player == "vlc":
                 # see https://wiki.videolan.org/Documentation:Format_String/, allow escaping with \$
                 self.title = self.title.replace("$", "$$").replace("\$$", "$")
                 extra_args.extend(["--input-title-format", self.title])

             # mpv
-            if self.cmd in SUPPORTED_PLAYERS['mpv']:
+            if self.player == "mpv":
                 # see https://mpv.io/manual/stable/#property-expansion, allow escaping with \$
                 self.title = self.title.replace("$", "$$").replace("\$$", "$")
                 extra_args.extend(["--title", self.title])    
```